src/ipc/ipc_client.ts

```diff
@@ -1327,4 +1327,27 @@ export class IpcClient {
   public cancelHelpChat(sessionId: string): void {
     this.ipcRenderer.invoke("help:chat:cancel", sessionId).catch(() => {});
   }
+
+  // --- Visual Editing ---
+  public async applyVisualEditingChanges(
+    changes: Array<{
+      componentId: string;
```

Comment:
you should avoid defining inline types/interfaces in this file instead define them in ipc_types.ts and share them across ipc_client and the respective handler, otherwise you'll get type definition inconsistencies between the client/server (which is what's happening here)